smg-automotive/eslint-config-pkg
# Initial setup for linter & formatter

References https://github.com/as24ch/au-docs/discussions/10

## Motivation and context

- Create a lean custom ESLint config for Linter, also add an initial setup for formatter using Prettier.
- Thanks to the original repo from CarForYou https://github.com/carforyou/carforyou-eslint-config-pkg

## How to test

- Simply run command `npm run test`.

```
info Run bad test cases with config from ./smg-au/eslint-config-pkg/src/__tests__/.eslintrc.js
info Expect to see 13 ESLint errors
info ESLint rules passed
info Run good test cases with config from ./smg-au/eslint-config-pkg/src/__tests__/.eslintrc.js
info Expect to see 0 ESLint errors
info ESLint rules passed
info Run bad test cases with config from ./smg-au/eslint-config-pkg/src/__tests__/.eslintrc.react.js
info Expect to see 21 ESLint errors
info ESLint rules passed
info Run good test cases with config from ./smg-au/eslint-config-pkg/src/__tests__/.eslintrc.react.js
info Expect to see 0 ESLint errors
info ESLint rules passed
info Run bad test cases with config from ./smg-au/eslint-config-pkg/src/__tests__/.eslintrc.next.js
info Expect to see 19 ESLint errors
info ESLint rules passed
info Run good test cases with config from ./smg-au/eslint-config-pkg/src/__tests__/.eslintrc.next.js
info Expect to see 0 ESLint errors
info ESLint rules passed
```
